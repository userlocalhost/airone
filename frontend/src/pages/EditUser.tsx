import { Box, Typography } from "@mui/material";
import React, { FC } from "react";
import { Link, useParams } from "react-router-dom";
import { useAsync } from "react-use";

import { topPath, usersPath } from "../Routes";
import { AironeBreadcrumbs } from "../components/common/AironeBreadcrumbs";
import { Loading } from "../components/common/Loading";
import { UserForm } from "../components/user/UserForm";
import { getUser } from "../utils/AironeAPIClient";

export const EditUser: FC = () => {
  const { userId } = useParams<{ userId: number }>();

  const user = useAsync(async () => {
    if (userId) {
      const resp = await getUser(userId);
      return await resp.json();
    }
  });

  return (
    <Box>
      <AironeBreadcrumbs>
        <Typography component={Link} to={topPath()}>
          Top
        </Typography>
        <Typography component={Link} to={usersPath()}>
          ユーザ管理
        </Typography>
        <Typography color="textPrimary">ユーザ編集</Typography>
      </AironeBreadcrumbs>

      {user.loading ? <Loading /> : <UserForm user={user.value} />}
    </Box>
  );
};
