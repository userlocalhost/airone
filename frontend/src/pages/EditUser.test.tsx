/**
 * @jest-environment jsdom
 */

import {
  render,
  waitForElementToBeRemoved,
  screen,
} from "@testing-library/react";
import React from "react";

import { TestWrapper } from "../utils/TestWrapper";

import { EditUser } from "./EditUser";

afterEach(() => {
  jest.clearAllMocks();
});

test("should match snapshot", async () => {
  // TODO avoid to pass the global variable implicitly
  (window as any).django_context = {
    user: {
      is_superuser: false,
    },
  };

  const user = {
    id: 1,
    username: "user1",
    email: "user1@example.com",
    is_superuser: false,
  };

  /* eslint-disable */
  jest.spyOn(require("../utils/AironeAPIClient"), "getUser").mockResolvedValue({
    json() {
      return Promise.resolve(user);
    },
  });
  /* eslint-enable */

  // wait async calls and get rendered fragment
  const result = render(<EditUser />, {
    wrapper: TestWrapper,
  });
  await waitForElementToBeRemoved(screen.getByTestId("loading"));

  expect(result).toMatchSnapshot();
});
