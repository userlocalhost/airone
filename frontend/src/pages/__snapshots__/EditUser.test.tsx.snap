// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <nav
          aria-label="breadcrumb"
          class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root makeStyles-breadcrumbs-1 css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
        >
          <ol
            class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
          >
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                href="/new-ui/"
              >
                Top
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <a
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                href="/new-ui/users"
              >
                ユーザ管理
              </a>
            </li>
            <li
              aria-hidden="true"
              class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
            >
              /
            </li>
            <li
              class="MuiBreadcrumbs-li"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
              >
                ユーザ編集
              </p>
            </li>
          </ol>
        </nav>
        <form>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            ユーザ編集
          </p>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-4 css-zcbmsk-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            保存
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <table
            class="MuiTable-root table table-bordered css-rqglhn-MuiTable-root"
          >
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    名前
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    メールアドレス
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  />
                </td>
              </tr>
              <tr
                class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    パスワード
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  <div
                    class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-a88p61-MuiInputBase-root-MuiInput-root"
                    required=""
                  >
                    <input
                      class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                      name="password"
                      required=""
                      type="password"
                      value=""
                    />
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root makeStyles-breadcrumbs-1 css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              href="/new-ui/"
            >
              Top
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              href="/new-ui/users"
            >
              ユーザ管理
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            /
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
            >
              ユーザ編集
            </p>
          </li>
        </ol>
      </nav>
      <form>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          ユーザ編集
        </p>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root makeStyles-button-4 css-zcbmsk-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          保存
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <table
          class="MuiTable-root table table-bordered css-rqglhn-MuiTable-root"
        >
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  名前
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                />
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  メールアドレス
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                />
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  パスワード
                </p>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                <div
                  class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary css-a88p61-MuiInputBase-root-MuiInput-root"
                  required=""
                >
                  <input
                    class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    name="password"
                    required=""
                    type="password"
                    value=""
                  />
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
